pkg/filter/remote_call_filter.go

```diff
@@ -31,40 +27,40 @@ import (
 
 import (
 	"github.com/dubbogo/dubbo-go-proxy/pkg/client"
-	"github.com/dubbogo/dubbo-go-proxy/pkg/client/dubbo"
 	"github.com/dubbogo/dubbo-go-proxy/pkg/common/constant"
 	"github.com/dubbogo/dubbo-go-proxy/pkg/common/extension"
 	"github.com/dubbogo/dubbo-go-proxy/pkg/context"
 	"github.com/dubbogo/dubbo-go-proxy/pkg/context/http"
 	"github.com/dubbogo/dubbo-go-proxy/pkg/logger"
+	"github.com/dubbogo/dubbo-go-proxy/pkg/pool"
 )
 
 func init() {
-	extension.SetFilterFunc(constant.HttpTransferDubboFilter, HttpDubbo())
+	extension.SetFilterFunc(constant.RemoteCallFilter, RemoteCall())
 }
 
-// HttpDubbo http 2 dubbo
-func HttpDubbo() context.FilterFunc {
+// RemoteCall http 2 dubbo
+func RemoteCall() context.FilterFunc {
 	return func(c context.Context) {
-		doDubbo(c.(*http.HttpContext))
+		doRemoteCall(c.(*http.HttpContext))
 	}
 }
 
-func doDubbo(c *http.HttpContext) {
+func doRemoteCall(c *http.HttpContext) {
 	api := c.GetAPI()
+	cl, e := pool.SingletonPool().GetClient(api.Method.IntegrationRequest.RequestType)
```

Comment:
defer pool.put to release the client?

Reply:
This client is singleton , May using same instance in different G will cause race condition?